test/sqllogictest/ct_liveness.slt

```diff
@@ -0,0 +1,167 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+mode cockroach
+
+simple conn=mz_system,user=mz_system
+ALTER SYSTEM SET enable_create_continual_task = true
+----
+COMPLETE 0
+
+# Start from a pristine state
+reset-server
+
+query T
+SHOW CLUSTER;
+----
+quickstart
+
+simple conn=mz_system,user=mz_system
+ALTER CLUSTER quickstart OWNER TO materialize
+----
+COMPLETE 0
+
+statement ok
+CREATE TABLE append_only (key INT, val STRING)
+
+statement ok
+CREATE CONTINUAL TASK upsert (key INT, val STRING) ON INPUT append_only AS (
+    DELETE FROM upsert WHERE key IN (SELECT key FROM append_only);
+    INSERT INTO upsert SELECT key, max(val) FROM append_only GROUP BY key;
+)
+
+statement ok
+INSERT INTO append_only VALUES (1, now()::STRING);
+
+query I
+SELECT COUNT(*) FROM upsert
+----
+1
+
+# The upsert CT and restarting the dataflow repeatedly turns out to be a great
+# way to shake out liveness issues without the full overhead of testdrive.
```

Comment:
This has popped up a few times already, I'll have to see how we can find those in an easier way.